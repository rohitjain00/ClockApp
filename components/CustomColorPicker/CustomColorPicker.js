import React, {useState} from 'react';
import {Button, Modal, StyleSheet, View} from 'react-native';
import {ColorPicker} from 'react-native-color-picker';
import PropTypes from 'prop-types';

export function CustomColorPicker(props) {
    const [modalVisible, setModalVisible] = useState(false);

    return (
        <View style={styles.container}>
            <Modal
                animationType="slide"
                transparent={false}
                visible={modalVisible}
                onRequestClose={() => {
                    setModalVisible(false);
                }}
            >
                <ColorPicker
                    onColorSelected={(color) => {
                        alert(`Color selected: ${color}`);
                        props.onChange(color);
                        setModalVisible(false);
                    }}
                    style={styles.colorPicker}
                />
                <Button
                    title="close"
                    onPress={() => {
                        setModalVisible(false);
                    }}
                />
            </Modal>
            <Button
                title={props.text}
                onPress={() => {
                    setModalVisible(true);
                }}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    colorPicker: {
        flex: 1
    },
    container: {
        marginTop: 20,
    },

});

CustomColorPicker.propTypes = {
    onChange: PropTypes.func,
    text: PropTypes.string
}