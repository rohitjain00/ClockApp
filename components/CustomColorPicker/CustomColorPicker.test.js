//CustomColorPicker.test.js
import React from 'react';
import {render} from '@testing-library/react-native';
import { CustomColorPicker } from "./CustomColorPicker";
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('CustomColorPicker Component', () => {
    const mockOnPress = jest.fn();
    const component = <CustomColorPicker  value={'#000000'} onChange={mockOnPress} text={'testing component'} />
    let wrapper;
    beforeEach(() => {
        wrapper = mount(component);
        jest.clearAllMocks();
    })
    it('Custom Color Picker Props Validation', async () => {
        const componentProps = wrapper.props();
        expect(componentProps.value).toEqual('#000000');
        expect(componentProps.onChange).toEqual(mockOnPress);
        expect(componentProps.text).toEqual('testing component');
    });
    it('Custom Color Picker Snapshot Validation', () => {
        const tree = render(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Custom Color Picker Action Validation', async () => {
        // console.log(wrapper.instance());
        // console.log(wrapper.childAt(1));
        // console.log(wrapper.state(['modalVisible']));
        // wrapper.childAt(1).onPressHandler();
        // expect(wrapper.state(['modalVisible'])).toEqual(true);
        // https://github.com/enzymejs/enzyme/issues/2263#issuecomment-568585047
        // According to above statement We cannot interact with state inside a hooks.
    });
    it('Custom Color Picker onChange Checker', async () => {
        wrapper.props().onChange('#eeeeee');
        expect(mockOnPress).toHaveBeenCalled();
        expect(mockOnPress).toHaveBeenCalledTimes(1);
    });
});