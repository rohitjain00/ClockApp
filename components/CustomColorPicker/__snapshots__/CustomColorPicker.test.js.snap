// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomColorPicker Component Custom Color Picker Snapshot Validation 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      />
      <View>
        <RCTSlider
          disabled={false}
          enabled={true}
          maximumValue={1}
          minimumValue={0}
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onSlidingComplete={null}
          onStartShouldSetResponder={[Function]}
          onValueChange={[Function]}
          step={0}
          style={
            Object {
              "height": 40,
            }
          }
          value={1}
        />
        <RCTSlider
          disabled={false}
          enabled={true}
          maximumValue={1}
          minimumValue={0}
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onSlidingComplete={null}
          onStartShouldSetResponder={[Function]}
          onValueChange={[Function]}
          step={0}
          style={
            Object {
              "height": 40,
            }
          }
          value={1}
        />
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          close
        </Text>
      </View>
    </View>
  </Modal>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        testing component
      </Text>
    </View>
  </View>
</View>
`;
